classes/submission/maps/Schema.php

```diff
@@ -105,11 +132,21 @@ public function map(Submission $item, LazyCollection $userGroups, array $genres)
      *
      * @param LazyCollection<int,UserGroup> $userGroups The user groups in this context
      * @param Genre[] $genres The file genres in this context
+     * @param ?Enumerable $reviewAssignments review assignments associated with a submission
+     * @param ?Enumerable $stageAssignments stage assignments associated with a submission
      */
-    public function summarize(Submission $item, LazyCollection $userGroups, array $genres): array
-    {
+    public function summarize(
+        Submission $item,
+        LazyCollection $userGroups,
+        array $genres,
+        ?Enumerable $reviewAssignments = null,
+        ?Enumerable $stageAssignments = null
+    ): array {
         $this->userGroups = $userGroups;
         $this->genres = $genres;
+        $this->reviewAssignments = $reviewAssignments ?? Repo::reviewAssignment()->getCollector()->filterBySubmissionIds([$item->getId()])->getMany();
```

Comment:
Here the `$this->reviewAssignments` is a `LazyCollection`, while `$this->stageAssignments` right bellow is a `Collection`. Do you think that we could take any advantage into changing the second one to `LazyCollection` by changing the `getStageAssignmentsBySubmissions`'s get function to `lazy()` or something similar?

Reply:
In theory - yes, it should be more memory efficient.